src/pip/_internal/network/auth.py

```diff
@@ -69,6 +70,7 @@ def get_keyring_auth(url, username):
         logger.warning(
             "Keyring is skipped due to an exception: %s", str(exc),
         )
+        keyring = None
```

Comment:
This is a pretty broad exception block to disable keyring as a part of, but with the warning we should get pretty fast feedback on what shouldn't be used as a criteria to skip.